spec/helpers.lua

```diff
@@ -49,7 +110,19 @@ function resty_http_proxy_mt:send(opts)
     opts.query = nil
   end
 
-  return self:request(opts)
+  local res, err = self:request(opts)
+  if res then
+    -- wrap the read_body() so it caches the result and can be called multiple times
+    local reader = res.read_body
+    res.read_body = function(self)
+      if (not self._cached_body) and (not self._cached_error) then
```

Comment:
I get that but I don't think they are useful there. 

> On Jun 24, 2016, at 12:22 AM, Thijs Schreijer <notifications@github.com> wrote:
> 
> In spec/helpers.lua:
> 
> > @@ -49,7 +110,19 @@ function resty_http_proxy_mt:send(opts)
> >      opts.query = nil
> >    end
> >  
> > -  return self:request(opts)
> > +  local res, err = self:request(opts)
> > +  if res then
> > +    -- wrap the read_body() so it caches the result and can be called multiple times
> > +    local reader = res.read_body
> > +    res.read_body = function(self)
> > +      if (not self._cached_body) and (not self._cached_error) then
> due to operator precedence, there can be very subtle bugs in a comparison involving and, or and not. Hence its good practice to use parenthesis to ensure it works according to the intentions and make it easier to read the code.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>